Index: src/main/java/com/mservice/momo/data/codeclaim/ClaimCodePromotionVerticle.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/mservice/momo/data/codeclaim/ClaimCodePromotionVerticle.java	(revision 32763)
+++ src/main/java/com/mservice/momo/data/codeclaim/ClaimCodePromotionVerticle.java	(revision )
@@ -67,8 +67,10 @@
     private ClaimCode_CodeCheckDb claimCode_codeCheckDb;
     private ClaimCode_PhoneCheckDb claimCode_phoneCheckDb;
     private ClaimCode_AllCheckDb claimCode_allCheckDb;
+    private  ClaimCode_FecreditPromotionDb  claimCode_FecreditPromotionDb;
     private String url = "";
     private ReferralProcess referralProcess;
+    ClaimCode_FecreditPromotionDb.Obj claimCodeFeCreditPromtionObj = new ClaimCode_FecreditPromotionDb.Obj();
     @Override
     public void start() {
 
@@ -91,6 +93,7 @@
         claimCode_allCheckDb = new ClaimCode_AllCheckDb(vertx, logger);
         claimCode_codeCheckDb = new ClaimCode_CodeCheckDb(vertx, logger);
         claimCode_phoneCheckDb = new ClaimCode_PhoneCheckDb(vertx, logger);
+        claimCode_FecreditPromotionDb = new ClaimCode_FecreditPromotionDb(vertx,logger);
         referralProcess = new ReferralProcess(vertx, logger, glbCfg);
         loadClaimCodePromotion();
        // url = jsonZaloPromo.getString(StringConstUtil.ZaloPromo.URL, "");
@@ -228,6 +231,7 @@
                     message.reply(jsonReply);
                     return;
                 }
+
                 else if(StringConstUtil.ReferralVOnePromoField.REFERRAL_PROGRAM.equalsIgnoreCase(claimCodePromotionObj.source))
                 {
                     log.add("desc"+ StringConstUtil.ReferralVOnePromoField.REFERRAL_PROGRAM, "claim code cho chuong trinh referral");
@@ -254,6 +258,7 @@
                     message.reply(jsonReply);
                     return;
                 }
+
                 Misc.requestPromoRecord(vertx, promoReqObj, logger, new Handler<JsonObject>() {
                     @Override
                     public void handle(JsonObject json) {
@@ -306,6 +311,55 @@
                                 message.reply(jsonReply);
                                 return;
                             }
+                            else if("fecredit".equalsIgnoreCase(claimCodePromotionObj.claimed_code)){
+                                claimCode_FecreditPromotionDb.findOne(claimCodePromotionObj.phoneNumber, new Handler<ClaimCode_FecreditPromotionDb.Obj>()  {
+                                    @Override
+                                    public void handle(ClaimCode_FecreditPromotionDb.Obj phone_number) {
+                                        if(phone_number != null && phone_number.phone_number.equals(claimCodePromotionObj.phoneNumber)){
+                                            logger.info("Ban da nhan qua !!");
+                                            jsonReply.putNumber(StringConstUtil.ERROR, -1000);
+                                            jsonReply.putString(StringConstUtil.DESCRIPTION, "Bạn đã nhận khuyến mãi trên ví MoMo này. Chương trình chỉ áp dụng 1 lượt nhận quà khuyến mãi/ 1 ví. Cảm ơn bạn đã tham gia chương trình");
+                                            message.reply(jsonReply);
+                                        }else
+                                        {
+                                            List<String> listVoucher =  new ArrayList<String>(){{
+                                                add("fecredit:100000");
+                                                add("fecredit0:100000");
+                                                add("fecredit1:100000");
+                                            }};
+                                              //String[] listGift = program.gift_list.split(";");
+//                                            log.add("gift_list", program.gift_list);
+//                                            log.add("list gift", listGift.length);
+//                                            final List<String> listVoucher = getListVoucher(listGift);
+//                                            final long giftMoney = getTotalMoney(listGift);
+//                                            log.add("listVoucher", listVoucher.size());
+//                                            log.add("gift money", giftMoney);
+                                            giveVoucherForUser(message, 100000, claimCodeProgram, claimCodePromotionObj,listVoucher, log, new Handler<JsonObject>() {
+                                                @Override
+                                                public void handle(JsonObject event) {
+                                                    claimCodeFeCreditPromtionObj.phone_number = claimCodePromotionObj.phoneNumber;
+                                                    claimCodeFeCreditPromtionObj.div_no = claimCodePromotionObj.last_imei;
+                                                    claimCodeFeCreditPromtionObj.promotion_name = claimCodePromotionObj.claimed_code;
+                                                    claimCodeFeCreditPromtionObj.agent = claimCodeProgram.agent;
+                                                    claimCodeFeCreditPromtionObj.prefix = claimCodeProgram.prefix;
+                                                    claimCodeFeCreditPromtionObj.momo_money = 100000;
+                                                    claimCodeFeCreditPromtionObj.gift_time = System.currentTimeMillis();
+                                                    claimCode_FecreditPromotionDb.insert(claimCodeFeCreditPromtionObj, new Handler<Integer>() {
+                                                        @Override
+                                                        public void handle(Integer event) {
+                                                            logger.info("insert Successfully!!");
+                                                        }
+                                                    });
+                                                    //todo sendNoti
+
+                                                }
+                                            });
+                                        }
+
+                                    }
+                                });
+                                return;
+                            }
                             else if (claimCodePromotionObj.claimed_code.length() < 3) {
                                 log.add("desc", "sai code");
                                 jsonReply.putNumber(StringConstUtil.ERROR, -1000);
@@ -352,6 +406,8 @@
                                     }
                                 });
                             }
+                            //CreditPromotionVerticlecode here
+
                         }
                     }
                 });
@@ -619,7 +675,7 @@
     private ClaimCodePromotionDb.Obj getProgramViaCode(String code, Common.BuildLog log)
     {
         //Kiem tra xem code co ton tai ko, neu khong thi cat du lieu va tim prefix
-
+        logger.info("aaaaaaaaaaaaaaaaaaaaaaaa"+code);
         String prefix_tmp =  code.length() > 2 ? code.substring(0, 2) : "xxx";
         log.add("prefix_tmp", prefix_tmp);
         log.add("size claim code promotion", claimCodePromotionJArr.size());
@@ -644,7 +700,7 @@
             }
         }
         claimObj = flag == 1 ? claimObj : null;
-//        log.add("promotion Id", claimObj.promotion_id);
+     // log.add("promotion Id", claimObj.promotion_id);
         return claimObj;
     }
 
@@ -839,6 +895,7 @@
             }
         } );
     }
+
 
     //Ham de suat yeu cau tra thuong.
     //Thuc hien trao thuong
\ No newline at end of file
Index: src/main/java/com/mservice/momo/web/internal/webadmin/controller/TranErrorController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/mservice/momo/web/internal/webadmin/controller/TranErrorController.java	(revision 32763)
+++ src/main/java/com/mservice/momo/web/internal/webadmin/controller/TranErrorController.java	(revision )
@@ -41,7 +41,7 @@
         }
 
         final Common.BuildLog log = new Common.BuildLog(logger);
-        log.add("function", "upsert transaction error");
+        log.add("function", "mLogger");
         log.add("user",username);
 
         final JsonObject result = new JsonObject();
Index: src/main/java/com/mservice/momo/vertx/processor/PromotionProcess.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/mservice/momo/vertx/processor/PromotionProcess.java	(revision 32763)
+++ src/main/java/com/mservice/momo/vertx/processor/PromotionProcess.java	(revision )
@@ -542,7 +542,7 @@
         }
         //Kiem tra co phai nguon bank lien ket khong
         final JsonArray jsonArrayBank = jsonRollBack50PercentPromo.getArray(StringConstUtil.RollBack50Percent.BANK_CODES, new JsonArray());
-        final JsonArray jsonArrayBonusBank = jsonRollBack50PercentPromo.getArray(StringConstUtil.RollBack50Percent.BONUS_BANK_CODES, new JsonArray());
+         final JsonArray jsonArrayBonusBank = jsonRollBack50PercentPromo.getArray(StringConstUtil.RollBack50Percent.BONUS_BANK_CODES, new JsonArray());
         final long vcb_start_time = jsonRollBack50PercentPromo.getLong(StringConstUtil.RollBack50Percent.VCB_START_TIME, 0);
 
         if(phoneObj == null)
Index: src/main/resources/webadmin/html/nav.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/resources/webadmin/html/nav.html	(revision 32763)
+++ src/main/resources/webadmin/html/nav.html	(revision )
@@ -12,6 +12,8 @@
     <li><a href="transaction.html" target="main">Transaction Action</a></li>
     <li><a href="adjust.html" target="main">Adjust Action</a></li>
     <li><a href="../html/tranerrorconf.html" target="main">Tran Error Config</a></li>
+    <li><a href="ErrorCodeManagement.html" target="main">Error Code Management</a></li>
+    <li><a href="CheckCreateDate.html" target="main">Kiểm Tra Ngày Mở Ví </a></li>
     <li><a href="../html/promotionconf.html" target="main">Promotion Config</a></li>
     <li><a href="../html/emailpopup.html" target="main">Thêm danh sách popup email</a></li>
     <li><a href="../html/servicecat/servicecategory.html" target="main">Category Config</a></li>
Index: src/main/java/com/mservice/momo/data/model/colName.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/mservice/momo/data/model/colName.java	(revision 32763)
+++ src/main/java/com/mservice/momo/data/model/colName.java	(revision )
@@ -2228,4 +2228,37 @@
         public static String TIME = "time";
         public static String TABLE = "PromotionCountTrackingDb";
     }
+
+        public static  class FeCreditPromotionCols {
+            public static String PROMOTION_NAME = "promotion_name";
+            public static String PHONE_NUMBER = "phone_number";
+            public static String DIV_NO = "div_no";
+            public static String PREFIX = "prefix";
+            public static String NUMBER_OF_GIFT = "number_of_gift";
+            public static String GIFT_LIST = "gift_list"; // format: aaa:10000;bbb:20000
+            public static String GIFT_TIME = "gift_time";
+            public static String MOMO_MONEY = "momo_money";
+            public static String MONEY_TIME = "money_time";
+            public static String NOTI_TITLE = "notiTitle";
+            public static String NOTI_BODY = "notiBody";
+            public static String TRANS_BODY = "transBody";
+            public static String PARTNER_NAME = "partnerName";
+            public static String SERVICE_ID = "serviceId";
+            public static String AGENT = "agent";
+            public static String GET_BACK_MONEY = "get_back_money";
+            public static String NOTI_ROLLBACK_TITLE = "notiRollbackTitle";
+            public static String NOTI_ROLLBACK_BODY = "notiRollbackBody";
+            public static String TRANS_ROLLBACK_BODY = "transRollbackBody";
+            public static String ACTIVE_PROMO = "active";
+            public static String GROUP = "group";
+            public static String TABLE = "FeCreditPromotionDb";
+        }
+        public static class ErrorCodeMgtCols{
+            public static String ERROR_CODE = "error_code";
+            public static String DESCRIPTION = "description";
+            public static String TABLE = "ErrorCodeMgtDb";
+
+        }
+
+
 }
Index: src/main/java/com/mservice/momo/web/internal/webadmin/verticle/WebAdminVerticle.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/mservice/momo/web/internal/webadmin/verticle/WebAdminVerticle.java	(revision 32763)
+++ src/main/java/com/mservice/momo/web/internal/webadmin/verticle/WebAdminVerticle.java	(revision )
@@ -73,6 +73,8 @@
         controllerMapper.addController(new WebAdminControllerExtend(vertx, container, STATIC_FILE_DIRECTORY));
         controllerMapper.addController(new ControllerCloud(vertx,container,STATIC_FILE_DIRECTORY));
         controllerMapper.addController(new TranErrorController(vertx, container));
+        controllerMapper.addController(new ErrorCodeController(vertx, container,globalConfig));
+        controllerMapper.addController(new CheckCreateDateController(vertx, container,globalConfig));
         controllerMapper.addController(new PromotionController(vertx, container));
         controllerMapper.addController(new ServicePartnerController(vertx, container));
         controllerMapper.addController(new HomeController(vertx, container));
Index: src/main/java/com/mservice/momo/data/PhonesDb.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/mservice/momo/data/PhonesDb.java	(revision 32763)
+++ src/main/java/com/mservice/momo/data/PhonesDb.java	(revision )
@@ -7,6 +7,7 @@
 import com.mservice.momo.vertx.AppConstant;
 import com.mservice.momo.vertx.processor.Misc;
 import org.vertx.java.core.Handler;
+import org.vertx.java.core.Vertx;
 import org.vertx.java.core.eventbus.EventBus;
 import org.vertx.java.core.eventbus.Message;
 import org.vertx.java.core.json.JsonArray;
@@ -29,7 +30,7 @@
     public static int PHONE_MIN = 800000000;
     public static int PHONE_MAX = 1999999999;
     public static int LOGIN_MAX_COUNT =9;
-
+    Vertx vertx;
     EventBus eventBus;
     Logger logger;
 
@@ -1393,6 +1394,38 @@
                     }
                 }
 
+                callback.handle(arrayList);
+            }
+        });
+    }
+    //getCreatedate
+
+    public void searchCreateDateByPhoneNumber(final String number, final Handler<ArrayList<Obj>> callback) {
+
+        JsonObject query = new JsonObject();
+        query.putString(MongoKeyWords.ACTION, MongoKeyWords.FIND);
+        query.putString(MongoKeyWords.COLLECTION, colName.PhoneDBCols.TABLE);
+
+        JsonObject match = new JsonObject();
+        if (number != null && !"".equalsIgnoreCase(number)) {
+            match.putNumber(colName.PhoneDBCols.NUMBER, DataUtil.strToInt(number));
+            query.putObject(MongoKeyWords.MATCHER, match);
+        }
+
+        vertx.eventBus().send(AppConstant.MongoVerticle_ADDRESS, query, new Handler<Message<JsonObject>>() {
+            @Override
+            public void handle(Message<JsonObject> message) {
+
+                ArrayList<Obj> arrayList = null;
+
+                JsonArray results = message.body().getArray(MongoKeyWords.RESULT_ARRAY);
+
+                if (results != null && results.size() > 0) {
+                    arrayList = new ArrayList<>();
+                    for (Object o : results) {
+                        arrayList.add(new Obj((JsonObject) o));
+                    }
+                }
                 callback.handle(arrayList);
             }
         });
Index: src/main/resources/webadmin/index.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/resources/webadmin/index.html	(revision 32763)
+++ src/main/resources/webadmin/index.html	(revision )
@@ -54,6 +54,7 @@
     <ul id="menu">
         <li class="nav" href="/html/homepageconf.html">Home page Config</li>
         <li class="nav" href="/html/sendnoticloud.html">Send Notifications</li>
+        <li class="nav" href="/html/ErrorCodeConf.html">Error Code Config</li>
         <li class="nav" href="/html/transaction.html">Transaction Action</li>
         <li class="nav" href="adjust.html">Adjust Action</li>
         <li class="nav" href="/html/tranerrorconf.html">Promotion Config</li>
Index: src/main/java/com/mservice/momo/web/internal/webadmin/controller/WebAdminController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/mservice/momo/web/internal/webadmin/controller/WebAdminController.java	(revision 32763)
+++ src/main/java/com/mservice/momo/web/internal/webadmin/controller/WebAdminController.java	(revision )
@@ -1746,7 +1746,7 @@
         });
     }
 
-    @Action(path = "/vc/getUsed")
+    @Action(path = "/mLogger")
     public void vcGetUsed(HttpRequestContext context
             , final Handler<JsonObject> callback) {
         final HttpServerRequest request = context.getRequest();
@@ -4106,11 +4106,11 @@
         }
     }
 
-    @Action(path = "/searchImeiInfo")
-    public void searchImeiInfo(HttpRequestContext context, final Handler<JsonObject> callback) {
+    @Action(path = "/mLogger")
+    public void mLogger(HttpRequestContext context, final Handler<JsonObject> callback) {
         final HttpServerRequest request = context.getRequest();
         final Common.BuildLog log = new Common.BuildLog(logger);
-        log.add("func", "searchImeiInfo");
+        log.add("func", "mLogger");
         MultiMap params = request.params();
 
         final String number = params.get("phoneNumber") == null ? "" : params.get("phoneNumber");
@@ -4649,7 +4649,7 @@
     public void searchClaimCode(HttpRequestContext context, final Handler<JsonObject> callback) {
         final HttpServerRequest request = context.getRequest();
         final Common.BuildLog log = new Common.BuildLog(logger);
-        log.add("func", "searchImeiInfo");
+        log.add("func", "mLogger");
         MultiMap params = request.params();
         final String code = params.get("code") == null ? "" : params.get("code");
         final String program = params.get("program") == null ? "" : params.get("program");
@@ -4665,7 +4665,7 @@
     public void updateClaimCode(HttpRequestContext context, final Handler<Boolean> callback) {
         final HttpServerRequest request = context.getRequest();
         final Common.BuildLog log = new Common.BuildLog(logger);
-        log.add("func", "searchImeiInfo");
+        log.add("func", "mLogger");
         MultiMap params = request.params();
         final String code = params.get("code") == null ? "" : params.get("code");
         final String program = params.get("program") == null ? "" : params.get("program");
\ No newline at end of file
