Index: src/main/java/com/mservice/momo/web/internal/webadmin/controller/PromotionController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/mservice/momo/web/internal/webadmin/controller/PromotionController.java	(revision 18174)
+++ src/main/java/com/mservice/momo/web/internal/webadmin/controller/PromotionController.java	(revision )
@@ -95,6 +95,8 @@
         obj.OFF_TIME_FROM = params.get("OffTimeFrom").trim();
         obj.OFF_TIME_TO = params.get("OffTimeTo").trim();
 
+        obj.ENABLE_PHASE2 = "true".equalsIgnoreCase(params.get("enablePhase2").trim()) ? true : false;
+
         log.add("params", obj.toJsonObject());
 
         if (!obj.isInvalid()) {
@@ -181,6 +183,7 @@
                 "  <th>Pin</th>" +
                 "  <th>OffTime From</th>" +
                 "  <th>OffTime To</th>" +
+                "  <th>Enable Phase 2</th>" +
                 "</tr>";
         for (int i = 0; i < objs.size(); i++) {
             result += getRowError(i, objs.get(i));
@@ -217,6 +220,7 @@
                 "  <td rid='" + position + "'>" + input.ADJUST_PIN + "</td>" +
                 "  <td rid='" + position + "'>" + input.OFF_TIME_FROM + "</td>" +
                 "  <td rid='" + position + "'>" + input.OFF_TIME_TO + "</td>" +
+                "  <td rid='" + position + "'>" + input.ENABLE_PHASE2+ "</td>" +
                 "</tr>";
         return result;
     }
\ No newline at end of file
Index: src/main/resources/webadmin/html/promotionconf.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/resources/webadmin/html/promotionconf.html	(revision 18174)
+++ src/main/resources/webadmin/html/promotionconf.html	(revision )
@@ -133,9 +133,14 @@
                     <div style="float:left;">
                         OffTime From<br>(yyyy-mm-dd hh:mm:ss): <input style="width:200px;border: 1px solid #778899;" type="text" id ="txtOffTimeFrom"> OffTime To(yyyy-mm-dd hh:mm:ss):  <input style="width:200px;border: 1px solid #778899;" type="text" id ="txtOffTimeTo">
                     </div>
-
                 </td>
             </tr>
+            <tr>
+                <td>Enable Phase 02</td>
+                <td>
+                    <input type="checkbox" id="chkEnable">
+                </td>
+            </tr>
 
             <tr>
                 <td colspan="2" align="center">
@@ -211,7 +216,12 @@
             $("#txtOffTimeFrom").val(tdArr[19].innerHTML);
             $("#txtOffTimeTo").val(tdArr[20].innerHTML);
 
-
+            var enablePhase2 = tdArr[21].innerHTML;
+            if("true" === enablePhase2 || true === enablePhase2){
+                $("#chkEnable").prop('checked', true);
+            }else{
+                $("#chkEnable").prop('checked', false);
+            }
         });
     });
 
@@ -257,7 +267,9 @@
                     id: $.session.get("id"),
                     username:$.session.get("username"),
                     OffTimeFrom:$("#txtOffTimeFrom").val(),
-                    OffTimeTo:$("#txtOffTimeTo").val()
+                    OffTimeTo:$("#txtOffTimeTo").val(),
+                    enablePhase2: $("#chkEnable").is(":checked"),
+
                 },
                 success: function (string) {
                     $('#mydiv').unmask();
@@ -284,6 +296,7 @@
                         $("#txtAdjustAccount").val("");
                         $("#txtOffTimeFrom").val("");
                         $("#txtOffTimeTo").val("");
+                        $("#chkEnable").prop('checked', false);
                         init();
                     }
                     else if (error == -1) {
\ No newline at end of file
Index: src/main/java/com/mservice/momo/vertx/vcb/VcbVerticle.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/mservice/momo/vertx/vcb/VcbVerticle.java	(revision 18174)
+++ src/main/java/com/mservice/momo/vertx/vcb/VcbVerticle.java	(revision )
@@ -351,11 +351,6 @@
     //give gift for viettinbank promotion.end
 
 
-
-
-
-
-
     //thuc hien khuyen mai PG galaxy
     private Handler<Message<JsonObject>> doPromotionForPG = new Handler<Message<JsonObject>>() {
         @Override
Index: src/main/java/com/mservice/momo/data/model/colName.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/mservice/momo/data/model/colName.java	(revision 18174)
+++ src/main/java/com/mservice/momo/data/model/colName.java	(revision )
@@ -488,6 +488,7 @@
 
         public static String OFF_TIME_FROM = "offtimefrom";
         public static String OFF_TIME_TO = "offtimeto";
+        public static String ENABLE_PHASE2 = "enablePhase2";
 
         public static String TABLE="promo";
     }
Index: src/main/java/com/mservice/momo/vertx/processor/TransProcess.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/mservice/momo/vertx/processor/TransProcess.java	(revision 18174)
+++ src/main/java/com/mservice/momo/vertx/processor/TransProcess.java	(revision )
@@ -2337,7 +2337,7 @@
             public void handle(JsonObject jsonObject) {
                 final PromotionDb.Obj glxRec = new PromotionDb.Obj(jsonObject);
 
-                //khong cho chuog trinh
+                //khong co chuog trinh
                 if (glxRec == null || glxRec.DATE_TO == 0  || glxRec.DATE_FROM == 0) {
                     log.add("galaxydesc", "Khong co chuong trinh khuyen mai nao");
                     log.writeLog();
@@ -2353,6 +2353,7 @@
                     return;
                 }
 
+                //khong phai rap galaxy
                 if(!"glx".equalsIgnoreCase(filmInfo.cinemaShortName)){
                     log.add("galaxydesc","Khong phai dat ve xem phim rap Galaxy");
                     log.writeLog();
@@ -2420,6 +2421,14 @@
                                         return;
                                     }
 
+                                    //chay phase 2, chi tra combo bap nuoc khi khach hang da mua ve xem phim lan dau
+                                    log.add("enable phase2",glxRec.ENABLE_PHASE2);
+                                    if(glxRec.ENABLE_PHASE2){
+                                        savedTime01Ok(msg, filmInfo, "Đã mua lần 1");
+                                        log.writeLog();
+                                        return;
+                                    }
+
                                     log.add("galaxydesc","tra voucher 30000d");
                                     //tang vc
                                     ArrayList<Misc.KeyValue> listKeyValue = new ArrayList<>();
@@ -2484,32 +2493,8 @@
                                                                     ,giftMessage
                                                                     ,transDb);
 
-                                                            //cap nhat so lan khuyen mai len 1
-                                                            Promo123PhimGlxDb.Obj saveObj = new Promo123PhimGlxDb.Obj();
-                                                            saveObj.ID = msg.cmdPhone +"";
-                                                            saveObj.TICKET_CODE = filmInfo.ticketCode;
-                                                            saveObj.INVOICE_NO = filmInfo.invoiceNo;
-                                                            saveObj.BUY_TIME = filmInfo.buyDate;
-                                                            saveObj.DISPLAY_TIME = filmInfo.displayTime;
-                                                            saveObj.NUMBER_OF_SEAT = filmInfo.seatAmount;
-                                                            saveObj.SEAT_LIST = filmInfo.seatList;
-                                                            saveObj.FILM_NAME = filmInfo.filmName;
-                                                            saveObj.RAP = filmInfo.cinemaFullName;
-                                                            saveObj.SUBRAP = filmInfo.cinemaSubName;
-                                                            saveObj.STATUS = Promo123PhimGlxDb.STATUS_NEW;
-                                                            saveObj.DESC = "Trả thưởng bằng voucher";
-                                                            saveObj.PROMO_COUNT = 1;
-                                                            saveObj.PROMO_TIMEVN = Misc.dateVNFormatWithTime(System.currentTimeMillis());
-                                                            saveObj.TIME = System.currentTimeMillis();
-                                                            saveObj.TIMEVN = Misc.dateVNFormatWithTime(System.currentTimeMillis());
-                                                            saveObj.AMOUNT = filmInfo.amount;
+                                                            savedTime01Ok(msg,filmInfo,"Trả thưởng bằng voucher");
 
-                                                            phim123Glx.save(saveObj, new Handler<Boolean>() {
-                                                                @Override
-                                                                public void handle(Boolean aBoolean) {
-
-                                                                }
-                                                            });
                                                             log.writeLog();
 
                                                         }else{
@@ -2654,6 +2639,46 @@
                         });
                     }
                 });
+            }
+        });
+    }
+
+
+    //ghi nhan da tra khuyen mai lan 1 roi
+
+    /***
+     * tracking the first time we did promotion for this wallet
+     * @param msg
+     * @param filmInfo
+     * @param desc
+     */
+    private void savedTime01Ok(MomoMessage msg
+            ,FilmInfo filmInfo
+            ,String desc) {
+        //cap nhat so lan khuyen mai len 1
+        Promo123PhimGlxDb.Obj saveObj = new Promo123PhimGlxDb.Obj();
+        saveObj.ID = msg.cmdPhone +"";
+        saveObj.TICKET_CODE = filmInfo.ticketCode;
+        saveObj.INVOICE_NO = filmInfo.invoiceNo;
+        saveObj.BUY_TIME = filmInfo.buyDate;
+        saveObj.DISPLAY_TIME = filmInfo.displayTime;
+        saveObj.NUMBER_OF_SEAT = filmInfo.seatAmount;
+        saveObj.SEAT_LIST = filmInfo.seatList;
+        saveObj.FILM_NAME = filmInfo.filmName;
+        saveObj.RAP = filmInfo.cinemaFullName;
+        saveObj.SUBRAP = filmInfo.cinemaSubName;
+        saveObj.STATUS = Promo123PhimGlxDb.STATUS_NEW;
+        saveObj.DESC = desc;
+        saveObj.PROMO_COUNT = 1;
+        saveObj.PROMO_TIMEVN = Misc.dateVNFormatWithTime(System.currentTimeMillis());
+        saveObj.TIME = System.currentTimeMillis();
+        saveObj.TIMEVN = Misc.dateVNFormatWithTime(System.currentTimeMillis());
+        saveObj.AMOUNT = filmInfo.amount;
+
+        phim123Glx.save(saveObj, new Handler<Boolean>() {
+            @Override
+            public void handle(Boolean aBoolean) {
+
             }
         });
     }
Index: src/main/java/com/mservice/momo/data/PromotionDb.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/com/mservice/momo/data/PromotionDb.java	(revision 18174)
+++ src/main/java/com/mservice/momo/data/PromotionDb.java	(revision )
@@ -216,6 +216,8 @@
         public String OFF_TIME_FROM="";
         public String OFF_TIME_TO="";
 
+        public boolean ENABLE_PHASE2=false;
+
         public Obj(){}
         public Obj(JsonObject jo){
 
@@ -271,6 +273,7 @@
 
             OFF_TIME_FROM = jo.getString(colName.PromoCols.OFF_TIME_FROM,"");
             OFF_TIME_TO = jo.getString(colName.PromoCols.OFF_TIME_TO,"");
+            ENABLE_PHASE2 = jo.getBoolean(colName.PromoCols.ENABLE_PHASE2,false);
 
         }
 
@@ -312,6 +315,7 @@
 
             jo.putString(colName.PromoCols.OFF_TIME_FROM,OFF_TIME_FROM);
             jo.putString(colName.PromoCols.OFF_TIME_TO,OFF_TIME_TO);
+            jo.putBoolean(colName.PromoCols.ENABLE_PHASE2,ENABLE_PHASE2);
 
             return jo;
         }
